### PR TITLE
feat: [M3-7011] - Add firewalls to search result queries

### PR DESCRIPTION
**Pull request by pmakode-akamai**
## Description 📝
Add firewalls to search result queries.

## Changes  🔄
- Added firewalls to search result queries.
- Changed Search bar text to "Search Products, IP Addresses, Tags…".
- Updated attribute names for Firewall:
   - Changed `created_dt` to `created`
   - Changed `updated_dt` to `updated`
- Updated e2e tests.
- Updated unit tests.

## Target release date 🗓️
N/A

## Preview 📷
Screenshot 2024-10-03 at 5 23 47 PM

## How to test 🧪

- Verify Search functionality for Firewalls work as expected.
- Verify changing these attribute names for Firewall does not break functionality in other places of the app.

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/api-v4/src/firewalls/types.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ export interface Firewall {
   label: string;
   tags: string[];
   rules: FirewallRules;
-  created_dt: string;
-  updated_dt: string;
+  created: string;
+  updated: string;
```
Comment (pmakode-akamai) [inline packages/api-v4/src/firewalls/types.ts:16]: <details>
<summary>API response for Firewalls</summary>

Screenshot 2024-10-03 at 5 56 54 PM
</details>

File: packages/manager/src/features/TopMenu/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -333,6 +334,7 @@ const SearchBar = (props: SearchProps) => {
           openMenuOnClick={false}
           openMenuOnFocus={false}
           options={finalOptions}
+          placeholder="Search Products, IP Addresses, Tags..."
```
Comment (pmakode-akamai) [inline packages/manager/src/features/TopMenu/SearchBar/SearchBar.tsx:337]: Changed search bar text based on comments from ticket M3-7011.

File: packages/api-v4/src/firewalls/types.ts (hunk 1 above)
Comment (bnussman-akamai) [reply packages/api-v4/src/firewalls/types.ts:16]: Nice fix. Thank you!

File: packages/manager/src/utilities/getFirewallDescription.ts
Comment (bnussman-akamai) [inline packages/manager/src/utilities/getFirewallDescription.ts:1]: What do you think about moving `getFirewallDescription` to `packages/manager/src/features/Firewalls/shared.ts`?

I'm not the biggest fan of having a single file dedicated to just one function. It looks like `packages/manager/src/features/Firewalls/shared.ts` is intended to be a place to hold general helper functions like this

**Review by bnussman-akamai — APPROVED** [review]

**Review by AzureLatte — APPROVED** [review]

File: packages/manager/src/utilities/getFirewallDescription.ts
Comment (dwiley-akamai) [reply packages/manager/src/utilities/getFirewallDescription.ts:1]: I am in favor of moving the function to `shared.ts` as well

**Review by dwiley-akamai — APPROVED** [review]
Functionality in Cloud looks good ✅
Checks pass ✅
Code review ✅

File: packages/manager/src/utilities/getFirewallDescription.ts
Comment (pmakode-akamai) [reply packages/manager/src/utilities/getFirewallDescription.ts:1]: I agree that moving `getFirewallDescription` to shared.ts is a good idea. 👍

**Pull request merged**